### PR TITLE
Add pull request template

### LINKED ISSUE #62: Add pull request template for PyDAX

### PR DESCRIPTION
**Pull request by xuhdev**
The imminent reason for having this template is to keep the ZenHub board
organized due to the confusion that both PRs and issues are on the same
board (while also informs potential contributors).

Close #62

---

## Checklist:

<details open>
<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #62

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [N/A] Add/update relevant tests.
- [N/A] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.

</details>

### CODE REVIEW

**Comment by xuhdev** [comment]
@nimicent I tagged you because you might be interested in the board management part.

File: .github/pull_request_template.md
Comment (edwardleardi) [inline .github/pull_request_template.md]:
```suggestion
  - [ ] No
  - [ ] Yes
```

File: .github/pull_request_template.md
Comment (edwardleardi) [inline .github/pull_request_template.md]:
```suggestion
  - [ ] No
  - [ ] Yes
```

File: .github/pull_request_template.md
Comment (edwardleardi) [inline .github/pull_request_template.md]: Cosmetically should we combine this into one line. The PR comment is going to be large, but not a huge concern.

Comment (xuhdev) [reply .github/pull_request_template.md]: How about the new version?

File: .github/pull_request_template.md
Comment (xuhdev) [inline .github/pull_request_template.md:12]: Not sure this should be singular or plural.

Comment (edwardleardi) [reply .github/pull_request_template.md:12]: Either probably works since sometimes a small PR is just one change

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md]: Don't think it addresses the cosmetic concern since everything is still on separate lines, but it's not a big deal. 

Who would have triage permission for this repo, just CODAIT team members?

Comment (xuhdev) [reply .github/pull_request_template.md]: No, currently only you, me and Julia have. We should probably give everyone the triage permission for scrum purposes.

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md:8]: This is to say that if the answer is no, this checkbox should also be checked.

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md]: I updated the permission. I don't know how to format these lines to be cosmetically reasonable though. I added a horizontal line at the top, which hopefully would be a good separator between the description of the PR and this checklist.

Comment (edwardleardi) [reply .github/pull_request_template.md]: How do you feel about enclosing the checklist in a dropdown so that it can be hidden when we don't need to look at the answers?

File: .github/pull_request_template.md
Comment (edwardleardi) [inline .github/pull_request_template.md:36]: Do you mean we should link the PR inside the issue or add the issue under `Linked Issues`(under the Milestone section of the PR)

File: .github/pull_request_template.md
Comment (edwardleardi) [inline .github/pull_request_template.md]: Can we add a section for this like `## PR Description:` below the `## Checklist:` section?

File: .github/pull_request_template.md
Comment (edwardleardi) [inline .github/pull_request_template.md:32]: How do you add a PR under a Zenhub epic? (maybe we should add a brief explanation here for those unfamiliar on how to do it too)

**Review by edwardleardi — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md]: oh I just saw your earlier comment about adding the line above `## Checklist:`. Maybe add `## PR Description:` above that line then?

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md]: Also by everyone do you mean the CODAIT team? We probably wouldn't use Zenhub to track external PR's right, or would we?

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md:36]: Yes, then they are tracked together

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md]: When you open new PR, the commit message should have been automatically inserted to the top. That's why I didn't put an additional section.

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md:32]: It's the same as issues, what text would you suggest?

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md]: Why not, it's a way to see how much contribution is from external people. We can do the ZenHub labor for them.

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md]: I updated the text. See if this would clear some confusion

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md:32]: Oh ok never mind, I know what happened now. I removed the Zenhub extension because it doesn't mesh well with GitHub dark mode, but as a result I don't see the Epic or Estimate options on PR's. My bad.

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md]: Oh ok gotcha. The text helps thanks. For me the commit message always goes into the PR title, but the description stays blank. I'll have to look into how Github handles this then.

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md:36]: Do you want to explicitly mention the `Linked Issues` section then, I think it would be clearer?

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md]: Gotcha, ok sounds good then.

Comment (edwardleardi) [reply .github/pull_request_template.md]: Did you see the message above about the dropdown?

> How do you feel about enclosing the checklist in a dropdown so that it can be hidden when we don't need to look at the answers?

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md:36]: It's called "Connect Issues", linked issues are GitHub's stuff, I think.

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md:32]: You don't have to do it on this page, the checklist is meant to be ticked after the PR is opened, so you would have an opportunity to head to the ZenHub user interface. You can't tick the boxes before the PR is opened anyway.

File: .github/pull_request_template.md
Comment (xuhdev) [reply .github/pull_request_template.md]: Sorry I missed that message. I added a dropdown. It seems like aesthetically this is the best I can do (you can't have the section header to the right of the triangle).

Comment (edwardleardi) [reply .github/pull_request_template.md]: Thanks, looks good

File: .github/pull_request_template.md
Comment (edwardleardi) [reply .github/pull_request_template.md:36]: oh ok I see it, it's on the bottom of the PR. The Linked Issues section also has the connected issue, maybe it does that automatically then after you use the Connect Issues section

**Review by edwardleardi — APPROVED** [review]
LGTM, not sure if @nimicent had a chance to look yet, so I'll let you merge whenever you think it's ready

**Comment by xuhdev** [comment]
I merged for now, if @nimicent wanna take a look, we can do a followup

**Pull request merged**
